src/objects.rs

```diff
@@ -1277,9 +1288,81 @@ pub struct ODRule {
     pub network_id: Option<String>,
     pub physical_mode_id: Option<String>,
 }
+impl_id!(ODRuleV1);
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[cfg(feature = "fare-v2")]
+pub struct Ticket {
+    pub id: String,
+    pub name: String,
+    pub comment: String,
+}
+#[cfg(feature = "fare-v2")]
 impl_id!(Ticket);
-impl_id!(ODRule);
+
+#[cfg(feature = "fare-v2")]
+impl GetObjectType for Ticket {
+    fn get_object_type() -> ObjectType {
+        ObjectType::Ticket
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[cfg(feature = "fare-v2")]
+pub struct TicketPrice {
+    pub ticket_id: String,
+    pub price: f64, // TODO: how to ensure this value is always positive (u64 on cents? Setter?)
+    pub currency: String, //TODO: evaluate crates 'iso4217' and 'steel-cent' for encoding currency
+    pub validity: ValidityPeriod,
```

Comment:
if you do this we need a Deserializer that transforms a  to `ticket_validity_start` and `ticket_validity_end` fields  to `ValidityPeriod` and vice versa

I suggest to have 2 fields for simplicity

Reply:
I didn't think about that. However, SerDe propose an functionality for that kind of problem [`#[serde(flatten)]`](https://serde.rs/attr-flatten.html). I'm going to try that instead (keep reusing our already defined `struct ValidityPeriod`). Done in 532b5cc84ca4cbadfeb85bdff5a0b68bc7907a65.

Reply:
So after, discussion, `#[serde(flatten)]` would help for flattening but would complexify the renaming. So I removed it in 1b2c1fa462d56b340ae84aa21aeeaa3f373a3f9c and replaced with 2 fields as proposed.